great_expectations/checkpoint/checkpoint.py

```diff
@@ -126,6 +131,107 @@ class Config:
+            by_alias=by_alias,
+            skip_defaults=skip_defaults,
+            exclude_unset=exclude_unset,
+            exclude_defaults=exclude_defaults,
+            exclude_none=exclude_none,
+            encoder=encoder,
+            models_as_dict=models_as_dict,
+        )
+
+        data = json.loads(json_data)  # Parse back to dict to add validation_definitions
+        data_with_validation_definitions = self._serialize_validation_definitions(data)
+        return json.dumps(data_with_validation_definitions, **dumps_kwargs)
+
+    @override
+    def dict(  # noqa: PLR0913
+        self,
+        *,
+        include: AbstractSet[int | str] | Mapping[int | str, Any] | None = None,
+        exclude: AbstractSet[int | str] | Mapping[int | str, Any] | None = None,
+        by_alias: bool = False,
+        skip_defaults: bool | None = None,
+        exclude_unset: bool = False,
+        exclude_defaults: bool = False,
+        exclude_none: bool = False,
+    ) -> Dict[str, Any]:
+        """
+        Override the default dict method to enable proper diagnostics around validation_definitions.
+
+        NOTE: This should be removed in favor of a field/model serializer when we upgrade to
+              Pydantic 2.
+        """
+        data = super().dict(
+            include=include,
+            exclude=self._determine_exclude(exclude=exclude),
+            by_alias=by_alias,
+            skip_defaults=skip_defaults,
+            exclude_unset=exclude_unset,
+            exclude_defaults=exclude_defaults,
+            exclude_none=exclude_none,
+        )
+
+        return self._serialize_validation_definitions(data=data)
+
+    def _determine_exclude(
+        self, exclude: AbstractSet[int | str] | Mapping[int | str, Any] | None
+    ) -> AbstractSet[int | str] | Mapping[int | str, Any] | None:
+        if not exclude:
+            exclude = set()
+
+        if isinstance(exclude, set):
+            exclude = exclude.add("validation_definitions")
+        else:
+            exclude["__all__"] = "validation_definitions"
+
+        return exclude
+
+    def _serialize_validation_definitions(self, data: dict) -> dict:
+        """
+        Manually serialize the validation_definitions field to avoid Pydantic's default
+        serialization.
+
+        We want to aggregate all errors from the validation_definitions and raise them as
+        a single error.
+        """
+        data["validation_definitions"] = []
+
+        diagnostics = CheckpointFreshnessDiagnostics(errors=[])
+        for validation_definition in self.validation_definitions:
+            try:
+                identifier_bundle = validation_definition.identifier_bundle()
+                data["validation_definitions"].append(identifier_bundle.dict())
+            except ResourceFreshnessAggregateError as e:
+                diagnostics.errors.extend(e.errors)
+
+        diagnostics.raise_for_error()
+        return data
```

Comment:
nit: Neither of these uses `self`, so they could be `class` or `static` methods.

Reply:
I believe the latter one does but good call on the former!